providers/logingov.go

```diff
@@ -270,12 +273,16 @@ func (p *LoginGovProvider) Redeem(ctx context.Context, _, code string) (*session
 }
 
 // GetLoginURL overrides GetLoginURL to add login.gov parameters
-func (p *LoginGovProvider) GetLoginURL(redirectURI, state, _ string) string {
+func (p *LoginGovProvider) GetLoginURL(redirectURI, state, _, codeChallenge, codeChallengeMethod string) string {
 	extraParams := url.Values{}
 	if p.AcrValues == "" {
 		acr := "http://idmanagement.gov/ns/assurance/loa/1"
 		extraParams.Add("acr_values", acr)
 	}
+	if codeChallenge != "" && codeChallengeMethod != "" {
+		extraParams.Add("code_challenge", codeChallenge)
+		extraParams.Add("code_challenge_method", codeChallengeMethod)
+	}
```

Comment:
This seems to be duplicated a lot across multiple providers, should it actually be a part of `makeLoginURL`?

Reply:
Moved!